File: bots/lab-examples/java/Spinner.java
import dev.robocode.tankroyale.botapi.*;
import dev.robocode.tankroyale.botapi.events.*;

/** Spinner — spin and fire. */
public class Spinner extends Bot {
    public static void main(String[] args) { new Spinner().start(); }
    @Override public void run() {
        while (isRunning()) { turnRight(30); turnGunRight(30); turnRadarRight(45); }
    }
    @Override public void onScannedBot(ScannedBotEvent e) { fire(2); }
    @Override public void onHitWall(HitWallEvent e) { back(30); turnLeft(90); }
}

File: bots/lab-examples/ts/Spinner.ts
import { Bot, HitWallEvent, ScannedBotEvent } from "@robocode.dev/tank-royale-bot-api";

/** Spinner — gira no lugar, radar sempre ligado, atira em tudo que aparecer. */
class Spinner extends Bot {
  static main() {
    new Spinner().start();
  }
  override run() {
    while (this.isRunning()) {
      this.turnRight(30);
      this.turnGunRight(30);
      this.turnRadarRight(45);
    }
  }
  override onScannedBot(_e: ScannedBotEvent) {
    this.fire(2);
  }
  override onHitWall(_e: HitWallEvent) {
    this.back(30);
    this.turnLeft(90);
  }
}
Spinner.main();
